client/src/pages/Home.tsx

```diff
@@ -0,0 +1,120 @@
+import React, {useEffect, useState} from 'react'
+import logo from './../logo.svg'
+import StoreList from './../components/StoreList'
+import './../App.css'
+
+interface UserUI {
+  id: string;
+  username: string;
+  name: string;
+  email: string;
+}
+
+function Home() {
+  // These interact with some global 'React' state? from the import React lib
+  // possibly, likely shared across different files in this running app
+  // welcome msg hook
+  const [welMsg, setWelMsg] = useState('')
+  // user query id
+  const [userid, setUserid] = useState('')
+  // users list?
+  const [usersList, setUsersList] = useState<UserUI[]>([])
+
+  // fetch welcome message
+  const fetchMsg = async () => {  // annonymous function that is async
+    let data = {
+      'id': userid
+    }
+    // Fetch /api
+    const msg = await fetch('/api/',{
+      method: 'POST',
+      mode: 'cors',
+      cache: 'no-cache',
+      credentials: 'include',
+      headers: {
+        'Content-Type': 'application/json'
+      },
+      redirect: 'follow',
+      referrerPolicy: 'no-referrer',
+      body: JSON.stringify(data)
+      })
+      .then(res => res.status === 200?res.text():"hello")  // same as passing 'res' to a function call
+                                // .then(func(res))
+                                // function func(res){ return res.text()) }
+    setWelMsg(msg)
+  }
+
+  const getUsers = (res: any) => {
+    let res_json = res.json()
+    if (res.ok){
+      return res_json
+    }
+    else{
+      alert(`Response code: ${res.status}`)
+      return []
+    }
+  }
+
+  const catchUsers = (err: any) => {
+    alert("Error: " + err)
+  }
+
+  // fetch list of users
+  const fetchUsers = async() => {
+    const users = await fetch('/api/users/all') // why the additional /all route?
+      .then(res => getUsers(res))  // returns the res as a json object
+      .catch(err => catchUsers(err))
+    setUsersList(users) // Is there something like runtime errors if json is not properly formatted?
+  }
+
+  // useEffect runs on rendered elements changing DOM update
+  useEffect(() => { // Named useEffect hooks? -> Likely custom react Hooks
+    fetchMsg()
```

Comment:
this fetched message everytime your state changed right?

Reply:
yes, I need to look into react state lifecycle model. currently too many unnecessary calls to api if we use this solution